src/mesh/analysis.rs

```diff
@@ -103,6 +103,38 @@ impl BoundingBox {
     pub fn diagonal_length(&self) -> f32 {
         nalgebra::distance(&self.minimum_point, &self.maximum_point)
     }
+
+    /// Check if the two bounding boxes intersect / share any portion of space
+
+    // https://math.stackexchange.com/questions/2651710/simplest-way-to-determine-if-two-3d-boxes-intersect
+    #[allow(dead_code)]
+    pub fn do_bounding_boxes_intersect(&self, other: &BoundingBox) -> bool {
```

Comment:
Now that this is a method, we can name it much better, e.g. `intersects` or `intersects_with`.